### ISSUE TITLE
Setting `PathBase` with an environment variable did not work

### ISSUE THREAD

**Issue by Alan-Kuan** [microsoft/fhir-server#3252]
**Describe the bug**
I set the value of an envrionment variable `FhirServer__PathBase` to `/api`, and deployed the FHIR Server on AKS.
However, when I tried to call the API endpoint like `{base_url}/api/Patient`, it responded me that the requested route was not found.

**FHIR Version?**
R4

**Data provider?**
SQL Server

**To Reproduce**
1. I added the following lines to `samples/kubernetes/helm/fhir-server/templates/deployment.yaml` under `spec.template.spec.containers.env`.
```yml
- name: FhirServer__PathBase
  value: "/api"
```
2. I installed the server with the following command at `samples/kubernetes`.
```sh
helm upgrade --install fhir-server ./helm/fhir-server/ \
    --create-namespace -n fhir-ns \
    --set database.dataStore=SqlContainer \
    --set database.sqlContainer.acceptEula=Y
```
3. I created an ingress with the following config.
```yml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: ingress-fhir
  namespace: fhir-ns
spec:
  ingressClassName: nginx
  rules:
  - host: 10.17.1.1.nip.io
    http:
      paths:
      - path: /api
        pathType: Prefix
        backend:
          service:
            name: fhir-server
            port:
              number: 80
```
4. I got the following response when I ran `curl http://10.17.1.1.nip.io/api/Patient | jq .`.
```json
{
    "resourceType": "OperationOutcome",
    "id": "0f90c6da-03c0-4dcd-9aef-cd7ad7177c8e",
    "issue": [
        {
            "severity": "error",
            "code": "not-found",
            "diagnostics": "The requested route was not found."
        }
    ]
}
```

**Expected behavior**
It should responded me with a bundle of `Patient` resources.

**Actual behavior**
It responded me with an `OperationOutcome` which told me the requested route was not found.

AB#102821

**Comment by EXPEkesheth** [comment]
@Alan-Kuan - When you forward the request to the FHIR server, please ensure you to use the {root of FHIR server url}.

**Comment by Alan-Kuan** [comment]
Hi.

Thanks for your reply.
A few weeks ago, I changed to configure the NGINX server inside the pod that served my frontend.
I made it also a reverse proxy that redirected requests under `/api` to the FHIR server.
I found the API server worked fine. It correctly included the `/api` in some urls in the response.

Thus, I guessed the original problem might caused by NGINX Ingress.
Since I found some features of it didn't work either on our AKS on Azure Stack HCI environment.
Though I still have not figured out the true reason.

I think this issue can be closed.
